source-google-forms contribution from bala-ceg

bala-ceg opened the pull request:
## What

This PR adds a new connector for source Google Forms (source-google-forms).

Contributor-provided connector description:

Google Forms is a free online tool from Google that allows users to create custom surveys, quizzes, and forms. It enables easy collection and organization of data by automating responses into a connected Google Sheets spreadsheet. With Google Forms, you can design forms with various question types, share them via email or links, and track responses in real-time, making it ideal for feedback collection, event registration, or educational assessments.
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `client_id` | `string` | Client ID.  |  |
| `client_secret` | `string` | Client secret.  |  |
| `client_refresh_token` | `string` | Refresh token.  |  |
| `form_id` | `array` | form_id.  |  |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| forms |  | No pagination | ✅ |  ❌  |
| form_responses | responseId | DefaultPaginator | ✅ |  ❌  |

</details>

## Reviewer checklist
- [ ] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

<!-- DO NOT REMOVE: connector-builder-new-connector-contribution -->

marcosmarxm commented on airbyte-integrations/connectors/source-google-forms/manifest.yaml:
```diff
@@ -0,0 +1,401 @@
+        type: SimpleRetriever
+        requester:
+          $ref: "#/definitions/base_requester"
+          path: "{{config[\"formId\"]}}"
```
Make this a paremeter request:
![image](https://github.com/user-attachments/assets/8329faad-18eb-480f-8518-62e8467d74a6)

marcosmarxm commented on airbyte-integrations/connectors/source-google-forms/manifest.yaml:
```diff
@@ -0,0 +1,401 @@
+        schema:
+          $ref: "#/schemas/forms"
+    form_responses:
+      type: DeclarativeStream
```
Include transformation to know from what form the response is from.

marcosmarxm reviewed (changes requested):
This is a very simple connector with 2 endpoints only but can be helpful to capture survey responses. Make sure it works with multiple forms.

marcosmarxm commented:
Moving next review to Nov 11th

bala-ceg commented:
@marcosmarxm, I addressed the review comments, added the icon.svg image and google forms api doesn't mention anything about pagination. Can you please review this connector ? Thanks
